.github/workflows/check-examples-grpc.yaml

```diff
@@ -57,14 +57,36 @@ jobs:
           toolchain: stable
           override: true
       - name: Create Astarte Cluster
-        id: astarte
         uses: astarte-platform/astarte-cluster-action@v1
         with:
-          astarte_version: "1.1.1"
+          astarte_version: "1.2.0"
       - name: Install interface
+        timeout-minutes: 3
         run: |
-          astartectl realm-management interfaces sync $GITHUB_WORKSPACE/astarte-device-sdk-python/examples/grpc/interfaces/*.json --non-interactive
-          astartectl realm-management interfaces ls
+          function install_interfaces() {
+            while true; do
+              echo "Installing the interfaces using astartectl"
+              astartectl realm-management interfaces sync $1/*.json --non-interactive
+              sleep 5
+              echo "Checking the installed interfaces."
+              installed_interfaces=$(astartectl realm-management interfaces ls)
+              missing_interface=false
+              for file in $1/*.json; do
+                interface_name=$(basename "$file" .json)
+                if ! grep -q "$interface_name" <<< "$installed_interfaces"; then
+                  echo "Error: Interface $interface_name not found in $installed_interfaces"
+                  missing_interface=true
+                fi
+              done
+              if [[ "$missing_interface" == "true" ]]; then
+                continue
+              else
+                echo "All interfaces have been installed correctly"
+                break
+              fi
+            done
+          }
+          install_interfaces $GITHUB_WORKSPACE/astarte-device-sdk-python/examples/grpc/interfaces
```

Comment:
What do you think to move the ` function install_interfaces` in a separate bash script?
So you can call the same script bash from different workflows, passing the interfaces to be installed.